lib/tlscontext.c

```diff
@@ -386,6 +413,10 @@ tls_context_setup_session(TLSContext *self)
         }
       else
         msg_debug("empty ssl options");
+
+      if (!tls_context_setup_ecdh(self->ssl_ctx, SN_X9_62_prime256v1))
+          goto error;
```

Comment:
Is the curve name always available? Why pass it as a parameter, as i see it is only used in one of the code paths, in other cases it is unused.